src/Subscription.ts

```diff
@@ -124,7 +124,7 @@ export class Subscription implements ISubscription {
    * `remove()` to remove the passed teardown logic from the inner subscriptions
    * list.
    */
-  add(teardown: TeardownLogic): Subscription {
+  add(teardown: TeardownLogic): Subscription | void {
```

Comment:
This is related with https://github.com/ReactiveX/rxjs/issues/1656, better to make `add` explicitly return subscriptions for all cases instead of allowing void, which closes issue.